### PR TITLE
Improve Command and Printer coupling issues, Favorite command improvements

### LINKED ISSUE #68: Add favoriting Cheatsheets Command

### PR DESCRIPTION
**Pull request by abnermtj**

### CODE REVIEW

File: src/main/java/parser/Parser.java [hunk 1]
```diff
@@ -39,14 +95,19 @@ private CommandEnum parseTypeOfCommand(String userInput) throws CommandException
...
         default:
             throw new CommandException("Please enter a valid command");
         }
     }
 
     private ArrayList<ArgumentFlagEnum> parseTypeOfArgument(String userInput) {
         ArrayList<ArgumentFlagEnum> argEnumList = new ArrayList<>();
-        Pattern pattern = Pattern.compile(" /[dnilk] ");
+
+        Pattern pattern = Pattern.compile(FLAG_REGEX);
```
Comment (theopin) [inline src/main/java/parser/Parser.java]: Perhaps a a better word than pattern for the variable name?

Comment (abnermtj) [reply src/main/java/parser/Parser.java]: alright fixed!

File: src/main/java/command/ViewCommand.java [hunk 2]
```diff
@@ -10,26 +10,28 @@
...
-            if (parser.getDescriptionMap().containsKey(ArgumentFlagEnum.NAME)) {
-                String name = parser.getDescriptionMap().get(ArgumentFlagEnum.NAME);
+            if (descriptionMap.containsKey(ArgumentFlagEnum.NAME)) {
+                String name = descriptionMap.get(ArgumentFlagEnum.NAME);
                 for (int i = 0; i < CheatSheetList.getSize(); i++) {
                     if (CheatSheetList.getCheatSheet(i).getCheatSheetName().equals(name)) {
                         index = i + 1;
                         break;
                     }
                 }
```
Comment (theopin) [inline src/main/java/command/ViewCommand.java]: Might wanna apply slap more here since iza little arrowheaded

Comment (abnermtj) [reply src/main/java/command/ViewCommand.java]: kk

**Review by brandontoh — APPROVED** [review]

**Pull request merged**